enterprise/templates/enterprise/grant_data_sharing_permissions.html

```diff
@@ -31,7 +31,11 @@
                         </h2>
                     </div>
                     <span class="text">
-                        ${_("In order to link your account, we need consent to share your course data with {provider}.").format(provider=sso_provider)}
+                        % if not course_id:
+                            ${_("In order to link your account, we need consent to share your course data with {provider}.").format(provider=ec_name)}
+                        % else:
+                            ${_("In order to access {course_name}, we need consent to share your course data with {provider}.").format(provider=ec_name, course_name=course_name)}
```

Comment:
@smarnach: In general, that'll be the case. However, in this particular case, the user has _already been enrolled_ in the course by the Enterprise admin. In cases where the user is manually enrolling in a course and has a coupon applied by the EnterpriseCustomer, there should be an option to opt-out.